api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -554,6 +554,10 @@ public List<NestedField> fields() {
       return lazyFieldList();
     }
 
+    public boolean containsField(int id) {
```

Comment:
Is this helpful? Further down, you use `schema.findField(field.sourceId()) != null` to check whether a schema contains a field (at any level) and that seems like a reasonable check to me.

Reply:
I am using it in `SparkPartitioningAwareScan`.

```
if (groupingKeyType().containsField(fieldId) && !seenFieldIds.contains(fieldId)) {
  groupingKeyFields.add(field);
}
```

I could use `groupingKeyType().field(fieldId) != null` but that seems harder to read. Especially, if the condition has multiple branches.

Reply:
Rewrote the other part and removed this.